pkg/gui/main_panel.go

```diff
@@ -22,6 +22,13 @@ func (gui *Gui) scrollDownMain(g *gocui.Gui, v *gocui.View) error {
 		_, sy := mainView.Size()
 		y += sy
 	}
+
+	_, sizeY := mainView.Size()
+	totalLines := mainView.ViewLinesHeight()
+	if oy+sizeY >= totalLines {
+		return nil
+	}
```

Comment:
I found a couple extra issues with this while testing. Mostly around the `gui.Config.UserConfig.Gui.ScrollPastBottom` flag. When that's set we should be able to continue scrolling until the main panel's buffer is empty. This minor alteration fixed the issues that I found :)

```
func (gui *Gui) scrollDownMain(g *gocui.Gui, v *gocui.View) error {
	mainView := gui.getMainView()
	mainView.Autoscroll = false
	ox, oy := mainView.Origin()

	reservedLines := 0
	if !gui.Config.UserConfig.Gui.ScrollPastBottom {
		_, sizeY := mainView.Size()
		reservedLines = sizeY
	}
	totalLines := mainView.ViewLinesHeight()
	if oy+reservedLines >= totalLines {
		return nil
	}

	return mainView.SetOrigin(ox, oy+gui.Config.UserConfig.Gui.ScrollHeight)
}
```

Reply:
Oh, i didn't realize there was such an option hahaha.
Ok, will put that in.